src/hydra-queue-runner/build-remote.cc

```diff
@@ -395,7 +445,8 @@ static void copyPathFromRemote(
           extractNarData(tee, localStore.printStorePath(info.path), narMembers);
       });
 
-      destStore.addToStore(info, *source2, NoRepair, NoCheckSigs);
+      store->addToStore(info, *source2, NoRepair, NoCheckSigs);
```

Comment:
re-adding to local store should be a no-op when it already has it: It is able skip bey looking at the path in `info`.

Reply:
@thufschmitt do you know why we need to add in both stores now? Does it have to do with realizations?

Reply:
I think it's an easy (and terribly inefficient :grimacing:) way of making sure that `localStore.queryRealisations` will succeed (which we IIRC we need later to get back some metadata about the path to put it in the db).

To be honest, we shouldn't do that but instead carry the needed metadata in memory until the point where we need it (but that can come later once we're confident that this is at least correct)

Reply:
OK thanks! I think that was my only question!

Reply:
Actually, since I want this PR to reach prod before the rest, I think it is better to kick this change out of here and back into the main PR.

Skipping this will be no problem for input-adressed derivations, and the point of this PR is, after all, to bridge the gap without causing regressions, not add new functionality.

Reply:
I think it has to be the ideal one to not mess up prod hydra, so I'll aim for that.